alacritty_terminal/src/tty/mod.rs

```diff
@@ -81,3 +78,48 @@ pub fn setup_env<C>(config: &Config<C>) {
         env::set_var(key, value);
     }
 }
+
+/// Check if a terminfo entry exists on the system.
+fn terminfo_exists(terminfo: &str) -> bool {
+    // Get first terminfo character for the parent directory.
+    let first = terminfo.get(..1).unwrap_or_default();
+    let first_hex = format!("{:x}", first.chars().next().unwrap_or_default() as usize);
+
+    // Return true if the terminfo file exists at the specified location.
+    macro_rules! check_path {
+        ($path:expr) => {
+            if $path.join(first).join(terminfo).exists()
+                || $path.join(&first_hex).join(terminfo).exists()
+            {
+                return true;
+            }
+        };
+    }
```

Comment:
I've decided to go with a macro here, since this completely avoids the allocation for the `Vec`, which is entirely unnecessary here.

So we can just easily check each path immediately and abort without pushing a bunch of useless paths to a vec that we never read anyways.